test/framework/cluster/state_validator.go

```diff
@@ -0,0 +1,70 @@
+package cluster
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+)
+
+// StateValidation defines a validation that can be registered to the StateValidator.
+type StateValidation = func(ctx context.Context, vc StateValidationConfig) error
+
+// RetriableStateValidation returns a StateValidation that is executed with the provided retrier.
+func RetriableStateValidation(retrier *retrier.Retrier, validation StateValidation) StateValidation {
+	return func(ctx context.Context, vc StateValidationConfig) error {
+		err := retrier.Retry(func() error {
+			return validation(ctx, vc)
+		})
+		return err
+	}
+}
+
+// StateValidator is responsible for checking if a cluster is valid against the spec that is provided.
+type StateValidator struct {
+	Config      StateValidationConfig
+	validations []StateValidation
+}
+
+// WithValidations registers multiple validations to the StateValidator that will be run when Validate is called.
+func (c *StateValidator) WithValidations(validations ...StateValidation) {
+	c.validations = append(c.validations, validations...)
+}
+
+// Validate runs through the set registered validations and returns an error if any of them fail after a number of retries.
+func (c *StateValidator) Validate(ctx context.Context) error {
+	errList := make([]error, 0)
+	for _, validate := range c.validations {
+		err := validate(ctx, c.Config)
+		if err != nil {
+			errList = append(errList, err)
+		}
+	}
+	return errors.NewAggregate(errList)
+}
+
+// Opt represents is a function that represents an option to configure a StateValidator.
+type Opt = func(cv *StateValidator)
```

Comment:
Any reason for the alias as opposed to definition? 

Could we also expand to `Option` leaving abbreviations primarily for variable names.

Reply:
Do we have `Option` implementations somewhere? If not we don't really need options. I also noted we accept the `StateValidationConfig` so users can just configure that as needed. Generally, options are only needed when configuring a set of defaults you want to expose configurability for.